src/doorman/TelnetProtocol.hpp

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include "Byte.hpp"
+
+#include <gsl/span>
+
+#include <vector>
+
+class TelnetProtocol {
+public:
+    struct Handler {
```

Comment:
I had to poke around for a bit to understand how TelnetProtocol's Handler callback mechanism gets wired up. But I see now:
- Channel implements Handler
- Channel composes Doorman & Xania
- Channel also has an instance of TelnetProtocol and in its construction, its passing a reference to itself (to its "Handler" interface really).

Reply:
Sorry it was confusing, it grew a little organically as these things do.

* Channel does indeed implement Handler. It implements it "privately" which means it's not a strict ISA relationship. No user of Channel can "see" that it implements Handler, one can't pass a Channel to something expecting a Handler. BUT Channel itself can pass itself as a handler. The fact that a channel implements an interface is an implementation internal. Another approach would be to have a set of delegates given to TelnetProtocol that all then defer onto various methods in Channel, but this is imo cleaner.
* Channel doesn't strictly compose Doorman and Xania: it has references to them both to be able to remove itself (Doorman interface) and send messages to the mud (Xania interface). In later PRs it might be nicer to literally pass in interfaces to both these things, then we could mock out and test `Channel`s.
* Channel does have a TelnetProtocol and passes itself to the handler interface exactly as you describe.